src/Bicep.Cli.IntegrationTests/DecompileParamsCommandTests.cs

```diff
@@ -0,0 +1,192 @@
+                File.ReadAllText(bicepparamPath).Should().BeEquivalentToIgnoringNewlines(expectedOutput);
+            }
+        }
+
+        [TestMethod]
+        public async Task Decompile_ValidParamFileWithBicepPath_ShouldSucceed()
+        {
+            var paramFile = 
+  @"{
+  ""$schema"": ""https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#"",
+  ""contentVersion"": ""1.0.0.0"",
+  ""parameters"": {
+    ""foo"": {
+      ""value"": ""bar""
+    }
+  }
+}";
+            var expectedOutput = 
+@"using './dir/main.bicep'
+
+param foo = 'bar'";
+
+            var (jsonPath, bicepparamPath) = Setup(TestContext, paramFile);
+
+            var (output, error, result) = await Bicep("decompile-params", jsonPath, "--bicep-file", "./dir/main.bicep");
+
+            using (new AssertionScope())
+            {
+                output.Should().BeEmpty();
+                error.AsLines().Should().Contain(DecompilationDisclaimer);
+                result.Should().Be(0);
+                File.ReadAllText(bicepparamPath).Should().BeEquivalentToIgnoringNewlines(expectedOutput);
+            }
+        }
+
+        [TestMethod]
+        public async Task Decompile_ValidParamFileWithStdOut_ShouldSucceed()
+        {
+            var paramFile = 
+  @"{
+  ""$schema"": ""https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#"",
+  ""contentVersion"": ""1.0.0.0"",
+  ""parameters"": {
+    ""foo"": {
+      ""value"": ""bar""
+    }
+  }
+}";
+            var expectedOutput = 
+@"using '' /*TODO: Provide a path to a bicep template*/
+
+param foo = 'bar'";
+
+            var (jsonPath, bicepparamPath) = Setup(TestContext, paramFile);
+
+            var (output, error, result) = await Bicep("decompile-params", jsonPath, "--stdout");
+
+            using (new AssertionScope())
+            {
+                output.Should().Be(expectedOutput);
+                error.AsLines().Should().Contain(DecompilationDisclaimer);
+                result.Should().Be(0);
+            }
+        }
+
+        [TestMethod]
+        public async Task Decompile_InvalidParamFile_ShouldFailWithErrors()
+        {
+            var paramFile = 
+  @"{
+  ""$schema"": ""https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#"",
+  ""contentVersion"": ""1.0.0.0"",
+  ""parameters"": {
+    ""foo"": { }
+  }
+}";
+
+            var (jsonPath, bicepparamPath) = Setup(TestContext, paramFile);
+
+            var (output, error, result) = await Bicep("decompile-params", jsonPath);
+
+            using (new AssertionScope())
+            {
+                output.Should().BeEmpty();
+                error.AsLines().Should().Contain(DecompilationDisclaimer);
+                error.AsLines().Should().Contain($"{jsonPath}: Decompilation failed with fatal error \"No value found parameter foo\"");
+                result.Should().Be(1);
+            }
+        }            
```

Comment:
I would move this to `ParamsDecompilationTests` since it's exercising decompiler behavior rather than CLI command behavior.

Reply:
Yes, this one is just to check if the error is properly propagated by the CLI